### PR TITLE
Fail the WebSocket if a pong isn’t received

### PR DESCRIPTION
**Pull request by ahulyk**
Fail the WebSocket if a pong isn’t received within some reasonable period #3227

### CODE REVIEW

File: okhttp/src/main/java/okhttp3/internal/ws/RealWebSocket.java [hunk 1]
```diff
@@ -131,6 +135,9 @@
   /** For testing. */
   int pongCount;
 
+  /** True if ping has been sent and pong is not received within some reasonable period.  */
+  private boolean pingSentPongNotRecived;
```
Comment (PromanSEW) [inline okhttp/src/main/java/okhttp3/internal/ws/RealWebSocket.java]: private boolean pingSentPongNotRec**e**ived;

Comment (ahulyk) [reply okhttp/src/main/java/okhttp3/internal/ws/RealWebSocket.java]: fixed

**Comment by swankjesse** [comment]
Thanks for this PR. I’ve written a slightly different PR that accomplishes the same goal.

**Comment by swankjesse** [comment]
https://github.com/square/okhttp/pull/3870

**Pull request closed without merging**